rubicon/objc/runtime.py

```diff
@@ -664,11 +664,35 @@ class objc_super(Structure):
 
 
 # http://stackoverflow.com/questions/3095360/what-exactly-is-super-in-objective-c
-def send_super(receiver, selName, *args, **kwargs):
+def send_super(cls, receiver, selName, *args, **kwargs):
     """Send a message named selName to the super of the receiver.
 
     This is the equivalent of [super selname:args].
+
+    Since proper 'super' semantics requires lexical scoping, the cls argument is now required.
+    It basically instructs this runtime what the lexical scope was of the caller, so that it
+    may find the appropriate superclass method implementation (if any) to call. (See issue #107)
+
+    As such, cls should be an instance of ObjCClass, the class of the lexical scope of the caller,
+    which is conveniently obtained simply by using the Python __class__ built-in (in the context of
+    an instance method implementation of a class descending from ObjCClass -- which all your custom
+    obj-c classes will always descend from).
+
+    Eg: send_super(__class__, self, 'init') would be the typical usage pattern.
+
+    cls may also be a naked Class pointer.
     """
+    if isinstance(cls, Class) or type(cls) is ObjCClass:
+        pass
+    else:
```

Comment:
Minor thing: this could be written as `if not isinstance(cls, (Class, ObjCClass))`, right?